src/backend/commands/tablecmds.c

```diff
@@ -19367,7 +19367,7 @@ ATPExecPartTruncate(Relation rel,
 
 		rv->location = pc->location;
 
-		if (prule->topRule->children)
+		if (prule->topRule && prule->topRule->children)
```

Comment:
I think this doesn't prevent the dereferencing [above](https://github.com/arenadata/gpdb/pull/738/files#diff-d373fb790605aded212c2e3a0fbe8c015d110077ff46c07d0a6d6647a2d7e928R19359).